### ISSUE TITLE
DSL2: Ensure that all sharded fastQs make it into mapping.

### ISSUE THREAD

**Issue by TCLamnidis** [nf-core/eager#1089]
(Preface: I might be under caffeinated and seeing things, BUT)

I think the sharded FastQs never actually make it to mapping? 
It looks like a channel `ch_input_for_mapping` is correctly created with them, but then all mappers in the if/else section of the MAP SWF start with: `ch_input_for_mapping = reads`, which pulls the original unsharded reads again... 🤔 

I also get the feeling that the mapping output files should actually contain the shard number, to ensure they are kept when merging lanes. 🤔 🤔

**Comment by TCLamnidis** [comment]
@nf-core/eager Could I get a second opinion? Am I missing something?

**Comment by shyama-mama** [comment]
Yes you are right we are using the original unsharded reads. We should be using `ch_input_for_mapping.reads` instead

**Comment by shyama-mama** [comment]
Made simple changes to fix the bug. 
https://github.com/nf-core/eager/pull/1090
